src/unix/bsd/netbsdlike/openbsd/mod.rs

```diff
@@ -540,12 +540,46 @@ impl siginfo_t {
         self.si_addr
     }
 
-    pub unsafe fn si_value(&self) -> ::sigval {
+    pub unsafe fn si_code(&self) -> ::c_int {
+        self.si_code
+    }
+
+    pub unsafe fn si_errno(&self) -> ::c_int {
+        self.si_errno
+    }
+
+    pub unsafe fn si_pid(&self) -> ::pid_t {
         #[repr(C)]
         struct siginfo_timer {
             _si_signo: ::c_int,
+            _si_code: ::c_int,
             _si_errno: ::c_int,
+            _pad: [::c_int; 29],
```

Comment:
sizeof(int) isn't a constant. it isn't the same value on i386 and amd64 for example.

I dunno if rust is able to get the sizeof(int) at build-time.

alternatively, you could define : 
```
pub SI_MAXSZ: ::c_int = 128;
pub SI_PAD: ::c_int = ((SI_MAXSZ / _sizeof_int) - 3);
```

and define a private `_sizeof_int` constant with the right value for each architecture supported.

Reply:
seems to work with the following commit

Reply:
please stick to the C definition.

something like the following should be possible:
```
const SI_MAXSZ: usize = 128;
const SI_PAD: usize = ((SI_MAXSZ / ::mem::size_of::<::c_int>()) - 3);
```

Reply:
oh no hold on my last commit did not go ...